sansio_lsp_client/client.py

```diff
@@ -285,3 +494,110 @@ def completions(
         if context is not None:
             params.update(context.dict())
         return self._send_request(method="textDocument/completion", params=params)
+
+    def hover(self, text_document_position: TextDocumentPosition) -> int:
+        assert self._state == ClientState.NORMAL
+        return self._send_request(
+            method="textDocument/hover", params=text_document_position.dict()
+        )
+
+    def signatureHelp(self, text_document_position: TextDocumentPosition) -> int:
+        assert self._state == ClientState.NORMAL
+        return self._send_request(
+            method="textDocument/signatureHelp", params=text_document_position.dict()
+        )
+
+    def definition(
+        self,
+        text_document_position: TextDocumentPosition,
+        # TODO PartialResultParams
+    ) -> int:
+        assert self._state == ClientState.NORMAL
+        return self._send_request(
+            method="textDocument/definition", params=text_document_position.dict()
+        )
+
+    def declaration(
+        self,
+        text_document_position: TextDocumentPosition,
+        # TODO PartialResultParams
+    ) -> int:
+        assert self._state == ClientState.NORMAL
+        return self._send_request(
+            method="textDocument/declaration", params=text_document_position.dict()
+        )
+
+    def typeDefinition(
+        self,
+        text_document_position: TextDocumentPosition,
+        # TODO PartialResultParams
+    ) -> int:
+        assert self._state == ClientState.NORMAL
+        return self._send_request(
+            method="textDocument/typeDefinition", params=text_document_position.dict()
+        )
+
+    def references(
+        self,
+        text_document_position: TextDocumentPosition,
+        # TODO PartialResultParams
+    ) -> int:
+        assert self._state == ClientState.NORMAL
+        params = {
+            "context": {"includeDeclaration": True},
+            **text_document_position.dict(),
+        }
+        return self._send_request(method="textDocument/references", params=params)
+
+    def call_hierarchy_in(self, text_document_position: TextDocumentPosition) -> int:
```

Comment:
I would prefer this to be named `prepareCallHierarchy` instead, so  that you can find it in LSP docs. Also applies to `doc_symbol` and `range_formatting`.

Reply:
`call_hierarchy_in()` - this is incomplete if I remember correctly. I was unable to find a server that supports it... should I remove it or indicate that is should not be used in some way?

Will rename the other ones.

Reply:
I'm fine with removing it or adding a TODO comment.